tests/ondemand/ondemand_object_tests.cpp

```diff
@@ -1074,8 +1074,111 @@ namespace object_tests {
     TEST_SUCCEED();
   }
 
+
+  bool test_strager() {
+    // Find exactly one of a/myint or b/myint, if both or none are present,
+    // return false.
+    auto process = [](ondemand::object root) -> std::pair<bool,uint64_t> {
+      ///////////////////////////////////////////////////////////
+      // The core idea with On-Demand is that you should visit everything only
+      // once.
+      //
+      // Thus, with On-Demand, you must consume a field once you have accessed it.
+      // Thus if you do root["a"], you must full consume the corresponding
+      // value (if present) before you move to another field (e.g., root["a"]).
+      //
+      // If, somehow, you only want the value corresponding to "a" if some
+      // other key is not present, then you must still parse/consume the value
+      // corresponding to "a", and later throw the result away.
+      ///////////////////////////////////////////////////////////
+      uint64_t myint;
+      // Notice in the line below that I use chaining... I do not
+      // first convert root["a"] to an object and then seek
+      // "myint" into. Chaining works in this instance because
+      // I only need to access the object 'root["a"]' for a single
+      // key, if I needed to do more work on root["a"], then I should
+      // create an object.
+      if(root["a"]["myint"].get(myint) == SUCCESS) {
```

Comment:
In my project, the code extracting `myint` is inside `handle_successful_response`, which is a virtual function. Calling `handle_successful_response` before validating the message is a problem. Moving the `myint` extraction into `process` is an abstraction violation.

Your proposed solution is undesirable for my project. I'd prefer a performance hit over an abstraction violation. But if I need to do it this way, I guess I have no choice.

Reply:
The following design will not work...

```C++
virtual void handle_successful_response(
    std::uint64_t request_id,
    simdjson::ondemand::value& result) = 0;
```

You need something like this... 

```C++
virtual void handle_successful_response(
    std::uint64_t request_id,
    myresulttype& result) = 0;
```